packages/twenty-server/src/engine/core-modules/workspace-invitation/services/workspace-invitation.service.ts

```diff
@@ -0,0 +1,125 @@
+import { InjectRepository } from '@nestjs/typeorm';
+import { ModuleRef } from '@nestjs/core';
+
+import { Repository, IsNull } from 'typeorm';
+
+import { Workspace } from 'src/engine/core-modules/workspace/workspace.entity';
+import { TokenService } from 'src/engine/core-modules/auth/services/token.service';
+import {
+  AppToken,
+  AppTokenType,
+} from 'src/engine/core-modules/app-token/app-token.entity';
+import { UserWorkspaceService } from 'src/engine/core-modules/user-workspace/user-workspace.service';
+
+// eslint-disable-next-line @nx/workspace-inject-workspace-repository
+export class WorkspaceInvitationService {
+  private tokenService: TokenService;
+  private userWorkspaceService: UserWorkspaceService;
+  constructor(
+    @InjectRepository(AppToken, 'core')
+    private readonly appTokenRepository: Repository<AppToken>,
+    private moduleRef: ModuleRef,
+  ) {}
+
+  onModuleInit() {
+    this.tokenService = this.moduleRef.get(TokenService, { strict: false });
+    this.userWorkspaceService = this.moduleRef.get(UserWorkspaceService, {
+      strict: false,
+    });
+  }
+
+  private async getOneWorkspaceInvitation(workspaceId: string, email: string) {
+    return await this.appTokenRepository
+      .createQueryBuilder('appToken')
+      .where('"appToken"."workspaceId" = :workspaceId', {
+        workspaceId,
+      })
+      .andWhere('"appToken".type = :type', {
+        type: AppTokenType.InvitationToken,
+      })
+      .andWhere('"appToken".context->>\'email\' = :email', { email })
+      .getOne();
+  }
+
+  private appTokenToWorkspaceInvitation(appToken: AppToken) {
```

Comment:
I'm not sure about the naming here
validateWorkspaceInvitationTokenOrThrow?

Reply:
Since we don't have a table invitation I use this function to convert an appToken from the database to a workspace invitation. It more like a casting than a validation. So I suggest `castAppTokenToWorkspaceInvitation`